### ISSUE TITLE
Model name "object" throws error

### ISSUE THREAD

**Issue by osteenbergen** [balderdashy/sails#3080]
Sails v 0.11.0

Just started with Sails and got the following error, took me a while to figure out its probably the api name that is causing it. I can run with other names just not "object"

```
sails new testproj
    info: Created a new Sails app `testproj`!
cd testproj/
sails generate api object
    info: Created a new api!
sails lift
```
```
module.js:161
  var exts = Object.keys(Module._extensions);
                    ^
TypeError: Object #<bound> has no method 'keys'
    at Function.Module._findPath (module.js:161:21)
    at Function.Module._resolveFilename (module.js:336:25)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Function.load [as cookieParser] (/usr/lib/node_modules/sails/node_modules/express/node_modules/connect/lib/connect.js:97:27)
    at /usr/lib/node_modules/sails/lib/router/index.js:109:39
    at /usr/lib/node_modules/sails/lib/app/private/after.js:91:14
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:251:17
    at done (/usr/lib/node_modules/sails/node_modules/async/lib/async.js:132:19)
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:32:16
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:248:21
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:572:34
    at handlerFn (/usr/lib/node_modules/sails/lib/app/private/after.js:78:13)
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:567:21
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:246:17
```

**Comment by osteenbergen** [comment]
Found another word: string

```
sails generate api string
```

```
util.js:35
  var str = String(f).replace(formatRegExp, function(x) {
            ^
TypeError: Property 'String' of object #<Object> is not a function
    at exports.format (util.js:35:13)
    at bound (/usr/lib/node_modules/sails/node_modules/lodash/dist/lodash.js:957:21)
    at eachController (/usr/lib/node_modules/sails/lib/hooks/blueprints/index.js:330:24)
    at forOwn (/usr/lib/node_modules/sails/node_modules/lodash/dist/lodash.js:2106:15)
    at Function.forEach (/usr/lib/node_modules/sails/node_modules/lodash/dist/lodash.js:3303:9)
    at Hook.bindShadowRoutes (/usr/lib/node_modules/sails/lib/hooks/blueprints/index.js:172:9)
    at bound (/usr/lib/node_modules/sails/node_modules/lodash/dist/lodash.js:729:21)
    at /usr/lib/node_modules/sails/lib/app/private/after.js:91:14
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:251:17
    at done (/usr/lib/node_modules/sails/node_modules/async/lib/async.js:132:19)
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:32:16
    at /usr/lib/node_modules/sails/node_modules/async/lib/async.js:248:21
    at Sails.<anonymous> (/usr/lib/node_modules/sails/node_modules/async/lib/async.js:572:34)
    at Sails.g (events.js:180:16)
    at Sails.EventEmitter.emit (events.js:117:20)
    at Sails.emitter.emit (/usr/lib/node_modules/sails/lib/app/private/after.js:50:11)
```

**Comment by bberry6** [comment]
@osteenbergen As far as I know, by default sails is putting models, services, etc in the global namespace.  So if the name of your model is a reserved word (like Object, String, etc), it will literally rewrite that reserved word with whatever the model is.

Try setting the options in config/global.js to `false`.

IMO, sails should have them default to false.  Even though it is convenient to have models and such in the global namespace, it is a bad practice which sails shouldn't encourage.

**Comment by tarlepp** [comment]
Yep, imho those globals should be false by default.

**Comment by osteenbergen** [comment]
I thought so, but i just renamed my model. Putting things in the global namespace should be discouraged.

**Comment by tjwebb** [comment]
> I thought so, but i just renamed my model. Putting things in the global namespace should be discouraged.

Calling a domain model `string` is only slightly less weird than calling it `constructor` or `prototype`. So then we'd have complains about `sails.models.prototype` being really confusing.

When you start naming stuff after reserved words in the language, there's just no way to win.

**Comment by sailsbot** [comment]
Thanks for posting, @osteenbergen.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
